data/valgrind/valgrind-test.c

```diff
@@ -0,0 +1,77 @@
+/* Valgrind test program */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+
+// memory pointers for still_reachable
+static void* mem_pointers[64];
+static int mem_pointer_i = 0;
+
+void say(const char* string) {
+	printf(string);
+}
+void say_hello() {
+	say("Hello World!\n");
+}
+
+void* leak_some_mem(size_t size) {
+	if(size == 0) {
+		errno = EINVAL;
+		return NULL;
+	}
+	void* mem = malloc(size);
+	return mem;
+}
+
+void printHelp(const char* progname) {
+	printf("Valgrind test program\n  Usage: %s [OPTIONS]\n\n", progname);
+	printf("OPTIONS\n\n");
+	printf("  --sayhello                Print hello (two function calls)\n");
+	printf("  --fork                    Fork to child (Use this as first argument!)\n");
+	printf("  --leak BYTES              Leak defined amount of memory\n");
+	printf("  --still-reachable BYTES   Leak defined amount of memory as still-reachable\n");
+	printf("\n");
+}
+
+int main(int argc, char** argv) {
+	int ret = EXIT_SUCCESS;
+	for (int i=1;i<argc;i++) {
+		const char *arg = argv[i];
+		if(strcmp("--help", arg) == 0 || strcmp("-h", arg) == 0) {
+			printHelp(argv[0]);
+			exit(EXIT_SUCCESS);
+		} else if(strcmp("--sayhello", arg) == 0) {
+			say_hello();
+		} else if(strcmp("--fork", arg) == 0) {
+			// Fork to child, wait for child to exit and return exit code of child
+			pid_t pid = fork();
+			if (pid < 0) {
+				fprintf(stderr, "Fork failed: %s\n", strerror(errno));
+				exit(EXIT_FAILURE);
+			} else if (pid > 0) {
+				pid = waitpid(pid, &ret, 0);
+				exit(ret);
```

Comment:
This line should be `exit(WEXITSTATUS(ret));`

It'd also be nice to add a test for use of uninitialized memory and maybe out-of-bounds read from stack.

Reply:
Hey Martin, thanks for looking at my PR - I've cahnged linke 58 accordingly

    				exit(EXIT_FAILURE);
    			} else if (pid > 0) {
    				pid = waitpid(pid, &ret, 0);
    				exit(WEXITSTATUS(ret));

and included the suggested tests for uninitialized memory and for out-of-bounds reads (heap and stack variables). The functions perform a simple xor hash over a memory range, that is either uninitialized or read out of the bounds of that memory.